### ISSUE TITLE
poor LOOCV R² performance of the GPR model

### ISSUE THREAD

**Issue by penguin1994** [Bin-Cao/Bgolearn#10]
Thank you for providing this excellent project.
I am very interested in your recent work, “Spatial-adaptive active learning identifies ultra-durable and highly active catalysts for acidic oxygen evolution reaction” (Sci. Bull., DOI: 10.1016/j.scib.2025.12.021).
I have been trying to reproduce the Gaussian Process Regression (GPR) results described in the Supplementary Information. According to Fig. S2a, the leave-one-out cross-validation (LOOCV) performance is reported as an R² of 0.58. However, when I use the data provided in the SI and perform LOOCV, I consistently obtain an R² of approximately −0.32, which is substantially lower than the reported value.

<img width="1304" height="1096" alt="Image" src="https://github.com/user-attachments/assets/da1c4aa2-4026-422f-be47-2f1afe47cff9" />

The code I used for the calculation is attached to this issue. 

[data.csv](https://github.com/user-attachments/files/31305375/data.csv)
[bgolearn_train.ipynb](https://github.com/user-attachments/files/31305374/bgolearn_train.ipynb)

I was wondering whether there were any additional preprocessing or model optimization steps that were not explicitly described.
Could you please clarify how the model was trained and evaluated for the LOOCV analysis?
If possible, it would also be very helpful if you could share the code.
Thank you very much!

**Comment by Bin-Cao** [comment]
[bgolearn_bincao.ipynb](https://github.com/user-attachments/files/31549608/bgolearn_bincao.ipynb)

**Comment by penguin1994** [comment]
Thanks! After updating Bgolearn to version 3.1.2, the LOOCV R² results are much better.
However, the R² values for each iteration are still:
[0.533085, 0.814124, 0.852183, 0.624180, 0.598593]
These values are still quite different from those shown in Fig. S2a ([0.58, 0.64, 0.67, 0.76, 0.82]), especially for iterations 2–4. 
In addition, I do not observe a clear improvement in R² across the iterations. Is this expected or normal?

[R2_each_iter.ipynb](https://github.com/user-attachments/files/31606976/R2_each_iter.ipynb)

By the way, could you please clarify how the MSE (mean squared error) in Fig. 2c was calculated? 

<img width="1723" height="513" alt="Image" src="https://github.com/user-attachments/assets/8c3a4320-9fff-41fc-b907-2e8f48159299" />

Is it directly obtained from the output of Bgolearn?
For example, for iteration 4, the LOOCV fitting results I obtained are:
```
Fitting goodness of LOOCV:
  RMSE = 32.867340   MAE = 25.024495   R2 = 0.598593
````
Can the MSE be calculated simply as:
MSE = RMSE²
If so, the resulting value seems quite large.
Also, Fig. 2c mentions that “Normalization is performed in each iteration.” I was wondering what type of normalization was applied and how it was implemented during each iteration.
Thank you very much for your help!

**Comment by Bin-Cao** [comment]
Thank you. The results reported in the Supplementary Materials are not directly exported from Bgolearn. In particular, previous versions of Bgolearn did not provide model-fitting results, because Bgolearn is primarily designed as a recommendation framework for searching for promising candidate experiments rather than as a tool for achieving the best possible fit to existing data. 

If a better-performing surrogate model is available, users can provide it to Bgolearn through the `Kriging_model` parameter. In the current version of Bgolearn, we additionally provide baseline model assessment to improve and evaluate the fitting performance.

Regarding the MSE, the reported value is calculated on normalized data and is therefore intended to reflect relative model performance. Both RMSE and MSE are calculated using the implementations in scikit-learn, following the corresponding definitions adopted by scikit-learn.

**Comment by Bin-Cao** [comment]
If you find our work helpful, we would greatly appreciate it if you could cite the papers:

* *Science Bulletin*, DOI: 10.1016/j.scib.2025.12.021
* *npj Computational Materials*, DOI: 10.1038/s41524-026-02226-3

Many thanks!
